src/generators/MongooseTypeGenerator.ts

```diff
@@ -0,0 +1,40 @@
+import FileWriter from "../FileWriter";
+import { mongoosePrimitiveTypes } from "../model/dataTypes/primitiveDataTypes";
+import classTemplate, {TemplateContext,} from "../templates/mongoose/classTemplate";
+import Generator from "./Generator";
+import FilePath from "../model/dataTypes/FilePath";
+import EntityDefinition from "../model/dataTypes/EntityDefinition";
+
+async function generate(
+  entityDefinition: EntityDefinition,
+  baseDirectory: FilePath
+): Promise<string> {
+  // TODO: work around primitive type names issue
+  let primitiveTypeName = entityDefinition.dataType.typeName.replace("Primitive", "").toLowerCase();
```

Comment:
Codacy found an issue: ['primitiveTypeName' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5738534)